### PR TITLE
Update readme names

### PR DESCRIPTION
**Pull request by danewalton**
Tending to @ahsonkhan comment here:
https://github.com/Azure/azure-sdk-for-c/pull/735#issuecomment-627107333

### CODE REVIEW

File: sdk/iot/README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Embedded C for Azure IoT Clients
+# Azure IoT Clients
 
 Azure SDK for Embedded C official IoT Hub client library.
```
Comment (ahsonkhan) [inline sdk/iot/README.md:3]: Is this accurate?

Comment (danewalton) [reply sdk/iot/README.md:3]: on line 3? I believe so. "Azure SDK for Embedded C" is the top level name so what follows seems reasonable

Comment (ahsonkhan) [reply sdk/iot/README.md:3]: Yes, I meant the other part `official IoT Hub client library.`

The title is `Azure IoT Clients`, but the tagline mentions `Hub`. Is that intentional?

This seems similar to the one in sdk/iot/hub/readme.md
`"Official Embedded C client for Azure IoT Hub.`

Comment (CIPop) [reply sdk/iot/README.md:3]: Good catch @ahsonkhan .  That is wrong: we have two separate services: Device Provisioning Service and IoT Hub. They are not the same and we shouldn't mix things up. The /iot folder will contain all IoT related clients.

Comment (CIPop) [reply sdk/iot/README.md:3]: Oh, I take that back: It is called Azure IoT Hub Device Provisioning Service.

File: sdk/iot/provisioning/readme.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Azure IoT Clients for Embedded C: Azure IoT Device Provisioning Client
+# Azure IoT Device Provisioning Client
```
Comment (CIPop) [inline sdk/iot/provisioning/readme.md:1]: We should correct here as well: https://docs.microsoft.com/en-us/azure/iot-dps/about-iot-dps

**Comment by ahsonkhan** [comment]
I am going to merge this now. Please follow up separately for any of the tag line changes.

**Pull request merged**